packages/dom/src/tabbable.js

```diff
@@ -133,3 +133,47 @@ export function find( context ) {
 		.map( mapObjectTabbableToElement )
 		.reduce( createStatefulCollapseRadioGroup(), [] );
 }
+
+/**
+ * Given an element, find the preceding tabbable elemnt.
+ *
+ * @param {Element} element The element before which to look.
+ */
+export function findPrevious( element ) {
+	const focusables = findFocusable( document.body );
```

Comment:
Why do we call / repeat the logic for `findFocusable`, rather than reuse `find` ? Is it a matter of optimizing to slice the set of focusables, before doing all the operations on the array?

Reply:
I guess we could find and filter first, then slice, but I thought it would be nice the make the optimisation. I don't think it matters much though, as it runs only once for a specific user action.

Reply:
I'm okay with the optimization. On a few occasions in the past, I'd wondered if we could refactor to optimize `find` better than we currently do. I could imagine pulling out some common function here to use across all the implementations which would accept an array of "pre-focusable-filtered" elements.

Reply:
Ok, I pulled out the common logic.